src/main.rs

```diff
@@ -1,3 +1,89 @@
-fn main() {
-    println!("Hello, World!");
+use std::fs;
+use anyhow::{Context, Result};
+use walkdir::WalkDir;
+
+mod arg;
+
+fn main() -> Result<()> {
+    let args = arg::parse_args();
+
+    println!("Cleaning directory: {:?}", args.dir);
+    println!("File types to clean: {:?}", args.types);
+    println!("Minimum file size: {} bytes", args.min_size);
+
+    directory_cleaner(&args.dir, &args.types, args.min_size)?;
+
+    println!("Cleaning completed successfully.");
+
+    Ok(())
 }
+
+
+fn directory_cleaner(dir: &String, paths_to_clear: &[String], size: u64) -> Result<()> {
+    // skip files that cannot be opened, due to permission issues etc
+    for entry in WalkDir::new(dir).into_iter().filter_map(|f| f.ok()) {
```

Comment:
why are we skipping files that can't be opened? Do we care about that? does the permission affect the deletion?

Reply:
Now updated in latest